packages/cdk/overlay/src/useOverlay.ts

```diff
@@ -0,0 +1,195 @@
+import type { Instance as PopperInstance } from '@popperjs/core'
+import type { ComputedRef } from 'vue'
+
+import type {
+  OverlayInstance,
+  OverlayOptions,
+  OverlayPopperEvents,
+  OverlayTrigger,
+  OverlayTriggerEvents,
+  RefElement,
+  TriggerElement,
+} from './types'
+
+import { ComponentPublicInstance, computed, reactive, ref, watch } from 'vue'
+import { createPopper } from '@popperjs/core'
+import { toRawType } from '@vue/shared'
+import { uniqueId } from '@idux/cdk/utils'
+
+import { usePopperOptions } from './usePopperOptions'
+
+const isHTMLElement = (val: unknown) => toRawType(val).startsWith('HTML')
+
+export const useOverlay = (options: OverlayOptions): OverlayInstance => {
+  const arrowRef = ref<RefElement>(null)
+  const triggerRef = ref<TriggerElement>(null)
+  const overlayRef = ref<RefElement>(null)
+  let popperInstance: Nullable<PopperInstance> = null
+
+  let showTimer: Nullable<number> = null
+  let hideTimer: Nullable<number> = null
+
+  const state = reactive(options)
+  const popperOptions = usePopperOptions(options, { arrow: arrowRef })
+
+  const overlayVisibility = (visible: boolean): HTMLElement | null => {
+    const overlayElement = overlayRef.value
+    if (overlayElement) {
+      overlayElement.style.display = visible ? 'block' : 'none'
+    }
+    return overlayElement
+  }
+
+  const initialize = (): void => {
+    const overlayElement = overlayVisibility(visibility.value)
+    if (!visibility.value) {
+      return
+    }
+    const unrefTrigger = triggerRef.value
```

Comment:
```suggestion
    const triggerElement = triggerRef.value
```
另外这里不需要判断 triggerElement  为不为空？